components/search/src/filter.rs

```diff
@@ -158,12 +160,52 @@ fn apply_overrides(engines: &mut [SearchEngineDefinition], overrides: &[JSONOver
     }
 }
 
+fn negotiate_languages(user_environment: &mut SearchUserEnvironment, available_locales: &[String]) {
+    let user_locale = user_environment.locale.to_lowercase();
+
+    let locales_map_to_en_us: HashSet<String> =
+        ENGLISH_LOCALES.iter().map(|s| s.to_lowercase()).collect();
+
+    let available_locales_set: HashSet<String> = available_locales
+        .iter()
+        .map(|locale| locale.to_lowercase())
+        .collect();
+
+    if available_locales_set.contains(&user_locale) {
+        return;
+    }
+    if locales_map_to_en_us.contains(&user_locale) {
+        user_environment.locale = "en-us".to_string();
+        return;
+    }
```

Comment:
I think we could simplify this to if user_locale starts with "en-" then make it "en-US".

The `en-*` locales we care about will presumably have already been listed in `available_locales_set`, so any other version of `en-` should just fall back to `en-US`. This would also avoid hard-coding the list of `en-*` locales which could easily be incomplete or be missed later.